src/authorization.js

```diff
@@ -525,13 +525,14 @@ class Authorization {
   }
 
   /**
-   * Sets the agent WebID for this authorization. Implemented as `setAgent()`
-   * setter method to enforce mutual exclusivity with `group` property, until
-   * ES6 setter methods become available.
+   * Sets the agent WebID for this authorization.
    * @method setAgent
-   * @param agent {String|Quad} Agent URL (or `acl:agent` RDF triple).
+   * @param agent {string|Quad|GroupListing} Agent URL (or `acl:agent` RDF triple).
    */
   setAgent (agent) {
```

Comment:
Another way to put this is that the current rules for mutating an `Authorization`'s agent/group is difficult to reason about. I think that part of the problem is that at a conceptual level, mutating a single authorization is a semantically broad operation. An `Authorization` is currently a tuple of modes, agent/group, allowed origins, etc. With the current set of setter methods you can modify an `Authorization` to the point that it's not remotely the same authorization anymore, e.g. by changing the access modes and the agent/group.